frontend/src/metabase/visualizations/echarts/cartesian/model/axis.ts

```diff
@@ -306,44 +312,162 @@ function calculateNonStackedExtent(
   });
 
   if (min === Infinity || max === -Infinity) {
-    return null;
+    return [0, 0];
   }
 
   return [min, max];
 }
 
 const NORMALIZED_RANGE: Extent = [0, 1];
 
-export function getYAxesExtents(
-  axisSplit: [DataKey[], DataKey[]],
-  data: GroupedDataset,
+const getYAxisFormatter = (
+  column: DatasetColumn,
   settings: ComputedVisualizationSettings,
-): AxisExtents {
-  if (data.length === 0) {
-    return [null, null];
+  renderingContext: RenderingContext,
+): AxisFormatter => {
+  const isNormalized = settings["stackable.stack_type"] === "normalized";
+
+  if (isNormalized) {
+    return (value: unknown) =>
+      renderingContext.formatValue(value, {
+        column,
+        number_style: "percent",
+      });
   }
 
-  const [leftAxisKeys, rightAxisKeys] = axisSplit;
+  const valueGetter = getMetricDisplayValueGetter(settings);
```

Comment:
Included formatters to the model with getMetricDisplayValueGetter right away to axis ticks correctly. This may be fixing bugs we did not find yet